src/covariances.jl

```diff
@@ -0,0 +1,87 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    design_matrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function design_matrix(comps)
+    matrix = reduce(hcat, components.(comps)[:,1])'
+    return matrix
+end
+
+"""
+    get_subcomposition_matrix(matrix, entries)
+
+Returns the subcomposition matrix given the `entries` (selected columns) from the `matrix`.
+"""
+function get_subcomposition_matrix(matrix, entries)
+    subcomposition_matrix = matrix[:, entries]
+    return subcomposition_matrix
+end
```

Comment:
This function isn't necessary to implement the covariance structures. Each covariance structure function should accept a vector of compositions and do loops manually instaed of allocating full design matrices.

Reply:
Function removed in commit ddb9f3f2dab7939210db1c56521e99b55952482d.